### ISSUE TITLE
Flaky tests due to ShadowParcel's nativeReadString, nativeSetDataPosition, nativeWriteInt...

### ISSUE THREAD

**Issue by ryansama** [robolectric/robolectric#5685]
Hi, I have posted a comment about this in #5017, but it seems to be inactive so I am posting a new one here.
### Description
Our tests sometimes fail due to a `NullPointerException` from `ShadowParcel`, for example: 
```
java.lang.NullPointerException
	at org.robolectric.shadows.ShadowParcel.nativeReadString(ShadowParcel.java:406)
	at android.os.Parcel.nativeReadString(Parcel.java)
	at android.os.Parcel$ReadWriteHelper.readString(Parcel.java:375)
	at android.os.Parcel.readString(Parcel.java:1996)
	at android.content.pm.ServiceInfo.__constructor__(ServiceInfo.java:129)
	at android.content.pm.ServiceInfo.<init>(ServiceInfo.java)
	at android.content.pm.ServiceInfo$1.createFromParcel(ServiceInfo.java:120)
	at android.content.pm.ServiceInfo$1.createFromParcel(ServiceInfo.java:118)
	at android.os.Parcel.readTypedObject(Parcel.java:2606)
	at android.os.Parcel.createTypedArray(Parcel.java:2568)
	at android.content.pm.PackageInfo.__constructor__(PackageInfo.java:505)
	at android.content.pm.PackageInfo.<init>(PackageInfo.java)
	at android.content.pm.PackageInfo$1.createFromParcel(PackageInfo.java:477)
	at android.content.pm.PackageInfo$1.createFromParcel(PackageInfo.java:474)
	at org.robolectric.shadows.ShadowApplicationPackageManager.newPackageInfo(ShadowApplicationPackageManager.java:417)
	at org.robolectric.shadows.ShadowApplicationPackageManager.getPackageInfo(ShadowApplicationPackageManager.java:393)
	at android.app.ApplicationPackageManager.getPackageInfo(ApplicationPackageManager.java)
	at ca.mycompany.app.MyCompanyApplication.onCreate(MyCompanyApplication.java:888)
	at org.robolectric.android.internal.AndroidTestEnvironment.lambda$installAndCreateApplication$0(AndroidTestEnvironment.java:276)
	at org.robolectric.util.PerfStatsCollector.measure(PerfStatsCollector.java:75)
	at org.robolectric.android.internal.AndroidTestEnvironment.installAndCreateApplication(AndroidTestEnvironment.java:276)
	at org.robolectric.android.internal.AndroidTestEnvironment.setUpApplicationState(AndroidTestEnvironment.java:169)
	at org.robolectric.RobolectricTestRunner.beforeTest(RobolectricTestRunner.java:301)
	at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$0(SandboxTestRunner.java:243)
	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:89)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

Note that at some point in the stack trace, our app is mentioned (`at ca.mycompany.app.MyCompanyApplication.onCreate(MyCompanyApplication.java:888)`), however when I place a breakpoint at that line, it does not pause the test. If it helps, this is the code at line 888: 
`final long currentVersion = getPackageManager().getPackageInfo(getPackageName(), 0).versionCode;`

Also please note that we have seen that the `NullPointerException` not only comes from `ShadowParcel.nativeReadString`, but also from others such as `ShadowParcel.nativeSetDataPosition`, and `ShadowParcel.nativeWriteInt`.

### Steps to Reproduce
This issue cannot be consistently reproduced on our CI or Android Studio.
### Robolectric & Android Version
Robolectric 4.3.1
Android Studio 4.0
Android Gradle Plugin 3.6.3 
compileSdkVersion 29
buildToolsVersion "29.0.3"
(please let me know if there is any more info you need)
### Link to a public git repo demonstrating the problem:
N/A

**Comment by juancamilo87** [comment]
Any thoughts or comment on this? This problem makes the tests flaky meaning that the CI tests are no longer reliable and are completely useless.

**Comment by hoisie** [comment]
Have you tried 4.4? It has some improvements to ShadowParcel.

**Comment by hoisie** [comment]
Let me know if these issues persist in 4.4 (or better yet 4.5-alpha-3) and I'll reopen. I am going to assume that the fixes in 4.4 addressed this.
